### PR TITLE
Reactrole refactor

### LINKED ISSUE #179: [BUG] Reactrole remove doesn't clear reaction
**Describe the bug**
Bot doesn't clear the reaction when removing it using `[p]reactrole remove`

**To Reproduce**
1. Run `[p]reactrole remove 392831042253488138-773843336434941963 💸 Finances`
2. The bot removes the reaction role from the config
3. The bot sends "React role removed."
4. The reaction is not cleared on the target message
5. The bot no longer responds to the aforementioned reaction

**Expected behavior**
In addition to the existing behaviour, the bot should clear the reaction it is removing on that message, if it successfully removes it from the config

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by issy**
# Initial Checklist
- [x] Has @tigattack / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [ ] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details
* **Does this resolve an issue?**
Resolves #179

### Features / Fixes
* Clears the target reaction in `[p]reactrole remove`
* Optimises some of the config logic in reactrole, in accordance with code in other cogs
* Added typehints to functions in reactrole

### Breaking Changes
N/A

## Additional
N/A

### CODE REVIEW

File: reactrole/reactrole.py [hunk 1]
```diff
@@ -85,77 +89,55 @@ async def add_reactrole(
...
         message: discord.Message,
-        reaction: str,
+        reaction: discord.Emoji,
         role: discord.Role,
     ):
         """Removes a configured react role
 
         Example:
-        - `[p]reactrole remove <message id> <reaction> <role>`
+        - `[p]reactrole remove 360678601227763712-893601663435276318 :kek: @moderator`
```
Comment (tigattack) [inline reactrole/reactrole.py:126]: Thanks Issy. Looks good from what I can tell.

Only suggestion is maybe it'd be useful to specify in help messages that the message ID should be specified as `<channel id>-<message id>` (at least that's what I'm assuming this is), or append this new example to the former example, but modify the former to something such as ``[p]reactrole remove <channel id>-<message id> <reaction> <role>``.

Thoughts?

Comment (issy) [reply reactrole/reactrole.py:126]: The bot will always generate a syntax hint for the command from the function arguments, so I feel like that would be a little redundant. This 'example' section serves only as a reminder of how those arguments would actually look in real-life usage. I think adequate information is provided there, but let me know if you think differently
image

Comment (tigattack) [reply reactrole/reactrole.py:126]: > The bot will always generate a syntax hint for the command from the function arguments

OK, sounds fine to me then 👍

**Review by tigattack — APPROVED** [review]
LGTMAMCSIAHTAAMTYVMI

**Pull request merged**